src/cli/plugin.ts

```diff
@@ -703,4 +704,129 @@ async function init() {
   log('Initialized', type.yellow, 'plugin', `${name}`.cyan, 'successfully!');
 }
 
-export default { install, update, check, init };
+function pluginLoaderFactory() {
```

Comment:
You should be able to make the `omegga` in the PluginLoader constructor optional:

```ts
  constructor(pluginsPath: string, omegga?: Omegga) {
    this.path = pluginsPath;
    this.omegga = omegga;
    this.store = Datastore.create({
      filename: path.join(omegga.dataPath, soft.PLUGIN_STORE),
      autoload: true,
    });
    this.store.persistence.setAutocompactionInterval(1000 * 60 * 5);
    this.formats = [];
    this.plugins = [];

    // soon to be deprecated !help
    omegga?.on('chatcmd:help', this.showHelp.bind(this));

    // use /plugins to get help
    omegga?.on('cmd:plugins', this.showHelp.bind(this));
  }
```

also you should be able to move the body of the .map.filter.map in PluginLoader.scan to a separate method called `scanPlugin(path: string)`

this line can be moved from `src/omegga/server.ts` to the PluginLoader's constructor:
    `this.pluginLoader.loadFormats(join(__dirname, 'plugin'));`

Reply:
may need to replace `pluginsPath` with just `workDir` and make these changes:

`this.path = path.join(workDir, soft.PLUGIN_PATH)`

`filename: path.join(workDir, soft.DATA_PATH, soft.PLUGIN_STORE)`